libs/portals/shared-modules/delegations/src/components/access/AccessCard.tsx

```diff
@@ -266,57 +254,61 @@ export const AccessCard = ({
               </Inline>
             </Box>
           )}
-          {showActions && (
+          {hasActions && (
             <Box
               display="flex"
               alignItems="center"
               justifyContent={['spaceBetween', 'flexEnd']}
               marginTop={[2, 0]}
               marginLeft={[0, 3]}
             >
-              {canDelete && (
+              {onDelete && (
                 <Button
                   variant="text"
                   icon="trash"
                   iconType="outline"
                   size="small"
                   colorScheme="destructive"
-                  onClick={() => onDelete(delegation)}
+                  onClick={() => onDelete?.(delegation)}
                   nowrap
                 >
                   {formatMessage(coreMessages.buttonDestroy)}
                 </Button>
               )}
               <Box marginLeft={3}>
-                {!isOutgoing && onView ? (
+                {onView && (
                   <Button
                     size="small"
                     variant="utility"
                     onClick={() => onView(delegation)}
                   >
                     {formatMessage(coreMessages.view)}
                   </Button>
-                ) : !isExpired && href ? (
+                )}
+                {!isExpired && onEdit ? (
                   <Button
                     icon="pencil"
                     iconType="outline"
                     size="small"
                     variant="utility"
-                    onClick={() => navigate(href)}
+                    onClick={() => onEdit(delegation)}
                   >
                     {formatMessage(coreMessages.buttonEdit)}
                   </Button>
-                ) : href ? (
-                  <Button
-                    icon="reload"
-                    iconType="outline"
-                    size="small"
-                    variant="utility"
-                    onClick={() => navigate(href)}
-                  >
-                    {formatMessage(coreMessages.buttonRenew)}
-                  </Button>
-                ) : null}
+                ) : (
+                  isExpired &&
+                  onRenew && (
+                    <Button
+                      icon="reload"
+                      iconType="outline"
+                      size="small"
+                      variant="utility"
+                      onClick={() => onRenew?.(delegation)}
```

Comment:
Unnecessary optional functional call chaining
```suggestion
                      onClick={() => onRenew(delegation)}
```